oceanwaves/spectral.py

```diff
@@ -29,14 +31,40 @@ def jonswap(f, Hm0, Tp, gamma=3.3, sigma_low=.07, sigma_high=.09,
         Method to compute alpha (default: yamaguchi)
     normalize : bool
         Normalize resulting spectrum to match ``Hm0``
-    
+
     Returns
     -------
     E : numpy.ndarray
-        Array of shape ``f`` with wave energy densities 
-    
+        Array of shape ``f, Hm0.shape`` with wave energy densities
+
     '''
 
+    # C Stringari - 04/06/2018
+    # check input data types to avoid the following error:
+    # ValueError: Integers to negative integer powers are not allowed.
+
+    # raise an warning if the frequency array starts with zero. if the
+    # user gives an array with zeros, the output will be inf at that
+    # frequency
+    if 0.0 in f:
+        warn("frequency array constains zeros.")
+
+    # get the input dtypes and promote to float 64, if needed
+    f = _check_dtype(f)
+    Hm0 = _check_dtype(Hm0)
+    Tp = _check_dtype(Tp)
+
+    # check shapes of Hm0 and Tp
+    if type(Hm0) == np.ndarray:
```

Comment:
It is better to check a variable type using *isinstance()*, because it will also match any inherited types. For example, the *OceanWaves* object is in fact a modified version of an *xarray.Dataset* object. If we would check a variable to be of type *xarray.Dataset* using *type()* it would fail on an *OceanWaves* object. Therefore we should use:

```
if isinstance(Hm0, np.ndarray):
```

here and in all similar if-statements below.